pkg/scheduler/nodeinfo/node_info.go

```diff
@@ -665,14 +665,20 @@ func (n *NodeInfo) FilterOutPods(pods []*v1.Pod) []*v1.Pod {
 			continue
 		}
 		// If pod is on the given node, add it to 'filtered' only if it is present in nodeInfo.
-		podKey, _ := GetPodKey(p)
+		var podKey string
+		var err error
+		if podKey, err = GetPodKey(p); err != nil {
+			klog.Errorf("Unable to get pod key: %v", err)
+		}
 		for _, np := range n.Pods() {
-			npodkey, _ := GetPodKey(np)
-			if npodkey == podKey {
-				filtered = append(filtered, p)
-				break
+			if npodkey, err2 := GetPodKey(np); err2 == nil {
+				if npodkey == podKey {
```

Comment:
I apologize for not taking a closer look at this earlier. If `GetPodKey` returns an error, then we should treat the key value as invalid. So we should not do this check if either call returned an error.